crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -630,22 +632,22 @@ unsafe impl<'a> WorldQuery for FilteredEntityRef<'a> {
         _table: &Table,
     ) {
         let mut access = Access::default();
-        state.access.component_reads().for_each(|id| {
-            if archetype.contains(id) {
+        for id in archetype.components() {
+            if state.access.has_component_read(id) {
                 access.add_component_read(id);
             }
-        });
+        }
         fetch.1 = access;
     }
 
     #[inline]
     unsafe fn set_table<'w>(fetch: &mut Self::Fetch<'w>, state: &Self::State, table: &'w Table) {
         let mut access = Access::default();
-        state.access.component_reads().for_each(|id| {
-            if table.has_column(id) {
+        for id in table.columns() {
+            if state.access.has_component_read(id) {
                 access.add_component_read(id);
             }
-        });
+        }
         fetch.1 = access;
```

Comment:
Yeah, the reason it was originally written like this was fixed in #13554.  

I think it can even be `fetch.1.clone_from(&state.access);` to make allocation here less likely.

Reply:
I made that change.